modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,317 @@
+        # Subsecond precision is not supported.
+        created = self.get_unix_time(int(now))
+        if self.signature_max_validity is not None:
+            expires = self.get_unix_time(now + self.signature_max_validity)
+
+        signed_headers_dict = {}
+        request_headers_dict = {}
+        for hdr_key in self.signed_headers:
+            hdr_key = hdr_key.lower()
+            if hdr_key == HEADER_REQUEST_TARGET:
+                value = request_target
+            elif hdr_key == HEADER_CREATED:
+                value = '{0}'.format(created)
+            elif hdr_key == HEADER_EXPIRES:
+                value = '{0}'.format(expires)
+            elif hdr_key == HEADER_DATE:
+                value = cdate
+                request_headers_dict[HEADER_DATE] = '{0}'.format(cdate)
+            elif hdr_key == HEADER_DIGEST:
+                request_body = body.encode()
+                body_digest, digest_prefix = self.get_message_digest(request_body)
+                b64_body_digest = b64encode(body_digest.digest())
+                value = digest_prefix + b64_body_digest.decode('ascii')
+                request_headers_dict[HEADER_DIGEST] = '{0}{1}'.format(
+                    digest_prefix, b64_body_digest.decode('ascii'))
+            elif hdr_key == HEADER_HOST:
+                value = target_host
+                request_headers_dict[HEADER_HOST] = '{0}'.format(target_host)
+            else:
+                value = headers[hdr_key]
+            signed_headers_dict[hdr_key] = value
+
+        return signed_header_dict, request_headers_dict
+
+    def get_message_digest(self, data):
+        """
+        Calculates and returns a cryptographic digest of a specified HTTP request.
+
+        :param data: The string representation of the date to be hashed with a cryptographic hash.
+        :return: A tuple of (digest, prefix).
+            The digest is a hashing object that contains the cryptographic digest of the HTTP request.
+            The prefix is a string that identifies the cryptographc hash. It is used to generate the
+            'Digest' header as specified in RFC 3230.
+        """
+        if self.signing_scheme in [SCHEME_RSA_SHA512, SCHEME_HS2019]:
+            digest = SHA512.new()
+            prefix = 'SHA-512='
+        elif self.signing_scheme in [SCHEME_RSA_SHA256]:
+            digest = SHA256.new()
+            prefix = 'SHA-256='
+        else:
+            raise Exception("Unsupported signing algorithm: {0}".format(self.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the signing_info.
+
+        :param digest: A hashing object that contains the cryptographic digest of the HTTP request.
+        :return: A base-64 string representing the cryptographic signature of the input digest.
+        """
+        self.load_private_key()
```

Comment:
Would the private key change between api calls?
How about moving this in to the `__init__` method.

Reply:
Yes, I can do that.